### PR TITLE
[SYCL][CUDA] Add cuda_dev_kit feature

### PR DESCRIPTION
**Pull request by steffenlarsen**
These changes add a cuda_dev_kit feature check for detecting availability of a CUDA SDK. CUDA include path and library path can be controlled through the new CUDA_INCLUDE and CUDA_LIBS_DIR variables.

### CODE REVIEW

File: SYCL/lit.cfg.py [hunk 1]
```diff
@@ -145,6 +146,29 @@
...
+check_cuda_file='cuda_include.cpp'
+with open(check_cuda_file, 'w') as fp:
+    fp.write('#include <cuda.h>\n')
+    fp.write('int main() { CUresult r = cuInit(0); return r; }')
+
+config.cuda_libs_dir=lit_config.params.get("cuda_libs_dir", config.cuda_libs_dir)
+config.cuda_include=lit_config.params.get("cuda_include", (config.cuda_include if config.cuda_include else config.sycl_include))
+
+cuda_options=cuda_options = (' -L'+config.cuda_libs_dir if config.cuda_libs_dir else '')+' -lcuda '+' -I'+config.cuda_include
+if cl_options:
```
Comment (AidanBeltonS) [inline SYCL/lit.cfg.py:166]: Should this be `cl_options` if this is for cuda?

**Review by AidanBeltonS — DISMISSED** [review]
LGTM

File: SYCL/lit.cfg.py (hunk 1 above)
Comment (steffenlarsen) [reply SYCL/lit.cfg.py:166]: Level zero uses this too. Seems like an obfuscated way of determining if it is on Windows.

Comment (AidanBeltonS) [reply SYCL/lit.cfg.py:166]: Okay, many thanks for clarifying.

**Review by v-klochkov — DISMISSED** [review]

**Review by v-klochkov — APPROVED** [review]

**Comment by steffenlarsen** [comment]
Build-log failure is unrelated and has since been fixed. Merging this.

**Pull request merged**